### ISSUE TITLE
Reduce size of libCaffe2.a

### ISSUE THREAD

**Issue by lhCheung1991** [facebookarchive/caffe2#1172]
Hi,

I wonder if there is any way to reduce the size of package libCaffe2.a. Does Caffe2 offer any method, like a customizing compilation, to select what you need in addition to the basic components?

Any suggestion will be appreciated.

Best Regards.

**Comment by lzx1413** [comment]
you can use the white.list to include the files you want to build in CMakeLists.txt

**Comment by lhCheung1991** [comment]
@lzx1413 
Thanks for your help!

I found that almost the unnecessary modules needed by the android package are switched off in the caffe2/scripts/build_android.sh. Does this script work for the whitelist? If so, is there any way more to reduce the package size for Android platform? 

Thanks again!

**Comment by lhCheung1991** [comment]
I am trying to deploy CNN model, like MobileNets, on Caffe2. And I removed the following components for the cmake system to reduce the size of libCaffe2.a.

1. SGD in trainning phase;
2. database reader & writer;
3. distributed trainning & inferencing operator;
4. gradient backprop operators;
5. loss function operators;
6. RNN operators

        deleted:    caffe2/db/create_db_op_gpu.cc
        deleted:    caffe2/db/db_test.cc
        deleted:    caffe2/db/leveldb.cc
        deleted:    caffe2/db/lmdb.cc
        deleted:    caffe2/db/rocksdb.cc
        deleted:    caffe2/db/zmqdb.cc
        deleted:    caffe2/distributed/file_store_handler_op_gpu.cc
        deleted:    caffe2/distributed/redis_store_handler.cc
        deleted:    caffe2/distributed/redis_store_handler_op.cc
        deleted:    caffe2/distributed/redis_store_handler_op_gpu.cc
        deleted:    caffe2/operators/accuracy_op.cc
        deleted:    caffe2/operators/apmeter_op.cc
        deleted:    caffe2/operators/batch_gather_ops.cc
        deleted:    caffe2/operators/communicator_op.cc
        deleted:    caffe2/operators/conv_gradient_op.cc
        deleted:    caffe2/operators/conv_op_cudnn.cc
        deleted:    caffe2/operators/conv_transpose_gradient_op.cc
        deleted:    caffe2/operators/conv_transpose_op.cc
        deleted:    caffe2/operators/conv_transpose_op_mobile.cc
        deleted:    caffe2/operators/counter_ops.cc
        deleted:    caffe2/operators/cross_entropy_op.cc
        deleted:    caffe2/operators/dataset_ops.cc
        deleted:    caffe2/operators/distance_op.cc
        deleted:    caffe2/operators/feed_blob_op.cc
        deleted:    caffe2/operators/gru_unit_op.cc
        deleted:    caffe2/operators/h_softmax_op.cc
        deleted:    caffe2/operators/index_ops.cc
        deleted:    caffe2/operators/instance_norm_gradient_op.cc
        deleted:    caffe2/operators/instance_norm_op.cc
        deleted:    caffe2/operators/load_save_op.cc
        deleted:    caffe2/operators/local_response_normalization_op.cc
        deleted:    caffe2/operators/loss_op.cc
        deleted:    caffe2/operators/lp_pool_op.cc
        deleted:    caffe2/operators/lstm_unit_op.cc
        deleted:    caffe2/operators/multi_class_accuracy_op.cc
        deleted:    caffe2/operators/one_hot_ops.cc
        deleted:    caffe2/operators/order_switch_ops.cc
        deleted:    caffe2/operators/pack_rnn_sequence_op.cc
        deleted:    caffe2/operators/perplexity_op.cc
        deleted:    caffe2/operators/pool_gradient_op.cc
        deleted:    caffe2/operators/prelu_op.cc
        deleted:    caffe2/operators/rank_loss_op.cc
        deleted:    caffe2/operators/recurrent_network_blob_fetcher_op.cc
        deleted:    caffe2/operators/recurrent_network_op.cc
        deleted:    caffe2/operators/recurrent_op_cudnn.cc
        deleted:    caffe2/operators/reduction_ops.cc
        deleted:    caffe2/operators/reverse_packed_segs_op.cc
        deleted:    caffe2/operators/roi_pool_op.cc
        deleted:    caffe2/operators/segment_reduction_op.cc
        deleted:    caffe2/operators/sequence_ops.cc
        deleted:    caffe2/operators/softmax_with_loss_op.cc
        deleted:    caffe2/operators/spatial_batch_norm_gradient_op.cc
        deleted:    caffe2/operators/spatial_softmax_with_loss_op.cc
        deleted:    caffe2/operators/stats_ops.cc
        deleted:    caffe2/operators/stop_gradient.cc
        deleted:    caffe2/operators/summarize_op.cc
        deleted:    caffe2/operators/tensor_protos_db_input.cc
        deleted:    caffe2/operators/top_k.cc
        deleted:    caffe2/operators/transpose_op.cc
        deleted:    caffe2/operators/utility_ops.cc
        deleted:    caffe2/operators/workspace_ops.cc
        deleted:    caffe2/operators/zero_gradient_op.cc
        deleted:    caffe2/perfkernels/embedding_lookup_avx2.cc
        deleted:    caffe2/perfkernels/typed_axpy_avx.cc
        deleted:    caffe2/perfkernels/typed_axpy_avx2.cc
        deleted:    caffe2/queue/queue_ops.cc
        deleted:    caffe2/queue/queue_ops_gpu.cc
        deleted:    caffe2/queue/rebatching_queue.cc
        deleted:    caffe2/queue/rebatching_queue_ops.cc
        deleted:    caffe2/sgd/adagrad_op.cc
        deleted:    caffe2/sgd/adam_op.cc
        deleted:    caffe2/sgd/ftrl_op.cc
        deleted:    caffe2/sgd/iter_op.cc
        deleted:    caffe2/sgd/iter_op_gpu.cc
        deleted:    caffe2/sgd/learning_rate_op.cc
        deleted:    caffe2/sgd/learning_rate_op_gpu.cc
        deleted:    caffe2/sgd/momentum_sgd_op.cc
        deleted:    caffe2/sgd/rmsprop_op.cc
